src/main/java/com/gepardec/wdg/challenge/ChallengesResource.java

```diff
@@ -62,15 +62,25 @@ public Response byId(@PathParam("id") @Min(value = 1) final Integer id) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Retry()
     public Response answer(@PathParam("id") @Min(value = 1, message = "{AnswerModel.id.min}") Integer id,
-            @NotNull(message = "{AnswerModel.notNull}") @Valid final Answer answer) {
+            @NotEmpty(message = "{AnswerModel.notNull}") @Valid final Answer answer) {
         log.info(String.format(Consts.INFO_003+" Provided Answer for challengeId='%s' with jobId='%s'", id, answer.getJobId()));
 
         final Challenges challenge = getChallengeForId(id);
         if (challenge == null) {
             log.warn(String.format(Consts.WARN_001+" Challenge with id='%s' with jobId='%s' not found!", id, answer.getJobId()));
             return buildChallengeNotFoundResponse(id);
         }
-        final boolean correctAnswer = challenge.getAnswer().trim().equalsIgnoreCase(answer.getAnswer().trim());
+
+        boolean correctAnswer;
+        URLValidator urlval = new URLValidator();
+
+        if (challenge.getId() == 2){
+            String urlPullRQ = answer.getAnswer().replace(" ", "");
+            log.info(String.format(Consts.INFO_006+" URL provided: url='%s''", urlPullRQ));
+            correctAnswer = urlPullRQ.substring(0, challenge.getAnswer().length()).matches(Challenges.CHALLENGE2.getAnswer()) && urlval.isValid(urlPullRQ, null) && urlPullRQ.substring(challenge.getAnswer().length()).matches(".*\\d.*");
+        } else {
+            correctAnswer = challenge.getAnswer().trim().equalsIgnoreCase(answer.getAnswer().trim());
+        }
```

Comment:
Komplettes refactoring der answer Methode, da das URL feld extra validiert werden muss für challenge 2.

- neue rest schnittstelle (methode) erstellen
- neue answer klasse erstellen (zusätzlich base klasse, von der answer für challenge 1 und 2 erben?)
- codeteile welche von answer-methode1 und answer-methode2 benutzt werden, in eigene methoden extrahieren, auf DUPLICATE-CODE achten!
- bei challenge2 kein answer feld benutzen, sondern URL und dort den URLValidator drauf hängen

Reply:
* DUPLICATE-CODE => Sorry, da sehe ich nicht so viele Möglichkeiten die Codeduplizierung zu vermeiden, da es meistens nur die Logs sind.
* Sonst, ich hoffe es würde so passen (wurde getestet und funktioniert alles). Falls ich noch was ändern soll, just write :D (vor allem das zweite POST und das Answer-splitting konnten anders umgesetzt werden - but its a start).